osutil/disks/disks.go

```diff
@@ -71,6 +74,30 @@ type Disk interface {
 	KernelDevicePath() string
 }
 
+// Partition represents a partition on a Disk device.
+type Partition struct {
+	// FilesystemLabel is the encoded filesystem label, this should only be
+	// compared with normal Go strings that are encoded with BlkIDEncodeLabel.
+	FilesystemLabel string
+	// FilesystemUUID is the encoded filesystem UUID, this should be compared
+	// with normal Go strings that are encoded with BlkIDEncodeLabel.
+	FilesystemUUID string
+	// PartitionLabel is the encoded partition label, this should only be
+	// compared with normal Go strings that are encoded with BlkIDEncodeLabel.
+	PartitionLabel string
+	// the partition UUID
+	PartitionUUID string
+	// Major is the major number for this partition.
+	Major int
```

Comment:
(nitpick^2) maybe having a pointer back to the `Disk` that the partition belongs to? But I guess this is fine.

Reply:
This could be useful eventually yes, I'll leave a TODO about it

Reply:
Added a TODO in https://github.com/snapcore/snapd/pull/10872